testgen/src/main/scala/TriangleTestGenerator.scala

```diff
@@ -0,0 +1,37 @@
+import java.io.File
+
+import testgen.TestSuiteBuilder._
+import testgen._
+
+object TriangleTestGenerator {
+  def main(args: Array[String]): Unit = {
+    val file = new File("src/main/resources/triangle.json")
+
+    def mapArgToString(arg: Any): String = {
+      arg match {
+        case vals: List[_] => vals.mkString(", ")
+        case _ => throw new IllegalStateException
+      }
+    }
+
+    def sutArgs(parseResult: CanonicalDataParser.ParseResult, argNames: String*): String =
+          argNames map (name => mapArgToString(parseResult(name))) mkString
+
+    def fromLabeledTest(argNames: String*): ToTestCaseData =
```

Comment:
I think for now, I am going to leave the test generator and the definition as is.

I like your more simplified generator.

```
val code = TestSuiteBuilder.build(file,
  fromLabeledTestAlt("equilateral" -> Seq("sides"), "isosceles" -> Seq("sides"),  "scalene" -> Seq("sides")))
```
But... The generator I wrote adds additional info to the test descriptions that I thinks is handy.
For example. 

```
  false if no sides are equal
``` 

becomes 

```
  isosceles - false if no sides are equal
```
when my generator runs. I don't think the original description is all that helpful. This is probably better to submit new descriptions to the canonical json. But, for now having the generator add the extra info to the beginning of the comments is useful IMHO.